### ISSUE TITLE
FALSE POSITIVE: `IL_INFINITE_LOOP` reports a float loop whose condition is false initially

### ISSUE THREAD

**Issue by lynae99** [spotbugs/spotbugs#4307]
Hi, I found a false positive in SpotBugs 4.10.3 when both sides of a loop comparison round to the same `float` value.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs IL_INFINITE_LOOP`

**Minimal reproducer**

```java
class SpotBugsSonarQubeILFloatRange {
    void test() {
        int start = 1_234_567_890;
        for (float value = start; value < start + 50; value++) {
        }
    }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-sonarqube-il-infinite-loop-fp-float-rounded-range.java
spotbugs -textui -effort:max -low -visitors InfiniteLoop classes
```

**Current behavior**

SpotBugs reports `IL_INFINITE_LOOP` at line 4.

**Expected behavior**

SpotBugs should not report this loop. Converting both comparison operands to `float` produces the same value, so the `<` condition is false before the first iteration.
